index_update.go

```diff
@@ -153,6 +153,13 @@ func compareMappings(ori, upd *mapping.IndexMappingImpl) error {
 		return fmt.Errorf("default datetime parser cannot be changed if index dynamic is true")
 	}
 
+	if ori.ScoringModel != upd.ScoringModel {
+		if ori.ScoringModel != "" && ori.ScoringModel != "tf-idf" && ori.ScoringModel != "bm25" ||
```

Comment:
There're constants in bleve_index_api for `tf-idf` and `bm25` which you should use here.

Also why check `ori.ScoringModel` here, just checking `upd.ScoringModel` should suffice yes?